misc/python/materialize/cli/ci_flake_detect.py

```diff
@@ -0,0 +1,116 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+#
+# ci_flake_detect.py - Detect flaky test failures based on comparing logs to patterns from Github issues
+
+import argparse
+import re
+import sys
+from typing import Any, List
+
+import junit_xml
+import requests
+
+from materialize import ci_util
+
+CI_RE = re.compile("^ci-regexp: (.*)$", flags=re.MULTILINE)
```

Comment:
This will match everything until the end of the text box, right? I think it is safer to not use multi-line but instead match everything from `ci-regexp:` to the end of the current line.

Reply:
No, ^ and $ are beginning and end of line still:
```
>>> CI_RE = re.compile("^ci-regexp: (.*)$", flags=re.MULTILINE)
>>> CI_RE.search("foo\nci-regexp: asdfgh\nbar\n")
<re.Match object; span=(4, 21), match='ci-regexp: asdfgh'>
```
But actually don't need multiline here, so let me remove it.